### ISSUE TITLE
DartPad is on an old (buggy) version of Dart2/Dart2JS

### ISSUE THREAD

**Issue by matanlurey** [dart-lang/dart-pad#840]
Including a version of Dart2JS that (accidentally) omits covariance checks.

You can see one such example of confusion here:
https://github.com/dart-lang/sdk/issues/33947

It would be very beneficial to update ASAP.

/cc @srawlins

**Comment by jcollins-g** [comment]
this is known.  currently blocked on dart-lang/dart-services#343

**Comment by srawlins** [comment]
Accidentally closed?

**Comment by matanlurey** [comment]
I realize this is functionally the same as https://github.com/dart-lang/dart-services/issues/343, but it's pretty confusing since I think nobody not on the Dart infrastructure team would know `dart-services` is the same as "DartPad".

(Maybe this is an argument to make a mono repo with all of the "dart infra" pieces, though?)

**Comment by jcollins-g** [comment]
Yep, it is confusing, which is why I referenced the issue it is a duplicate of and marked this as "duplicate" before closing it.

**Comment by srawlins** [comment]
Oops, didn't see the duplicate tag. Thanks!

**Comment by jcollins-g** [comment]
FYI:  https://jcollins1-debug-dot-dart-pad.appspot.com/ is pointing to a dart-services backend with dart-lang/dart-services#344, if you'd like to try it.  Blocking issues finally cleared out of the way.
